src/view/uniprotkb/components/XRefView.tsx

```diff
@@ -85,24 +90,84 @@ type XRefProps = {
   crc64?: string;
 };
 
+const EMBLXref: React.FC<{
+  databaseInfo: DatabaseInfoPoint;
+  params: { [key: string]: string };
+  id: string | undefined;
+  xref: Xref;
+  isoformNode?: JSX.Element;
+}> = ({ databaseInfo, params, id, xref, isoformNode }) => {
+  // M28638 (EMBL|GenBank|DDBJ)
+  const genBankInfo = databaseToDatabaseInfo.GenBank;
+  const ddbjInfo = databaseToDatabaseInfo.DDBJ;
+  const { properties, additionalIds } = xref;
+  if (!genBankInfo || !ddbjInfo) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'GenBank or DDBJ database information not found in drlineconiguration'
+    );
+  }
+  return (
+    <Fragment>
+      (
+      <ExternalLink url={processUrlTemplate(databaseInfo.uriLink, params)}>
+        EMBL
+      </ExternalLink>
+      {' | '}
+      <ExternalLink url={processUrlTemplate(genBankInfo.uriLink, params)}>
+        GenBank
+      </ExternalLink>
+      {' | '}
+      <ExternalLink url={processUrlTemplate(ddbjInfo.uriLink, params)}>
+        DDBJ
+      </ExternalLink>
+      {') '}
+      {id && <ExternalLink url={externalUrls.ENA(id)}>{id}</ExternalLink>}
+      {additionalIds &&
+        additionalIds.map(additionalId => (
+          <ExternalLink url={externalUrls.ENA(additionalId)} key={additionalId}>
+            {additionalId}
+          </ExternalLink>
+        ))}
+      {properties &&
+        properties.MoleculeType &&
+        `${
+          ((EMBLXrefProperties as unknown) as { [key: string]: string })[
```

Comment:
can avoid the use of `uknown` with:
```
EMBLXrefProperties[properties.MoleculeType as keyof typeof EMBLXrefProperties
```
and likewise at L145

Reply:
Nice.